src/util/getThoughtCommands.ts

```diff
@@ -0,0 +1,85 @@
+import CommandState from '../@types/CommandState'
+import FormattingCommand from '../@types/FormattingCommand'
+
+const commands = Object.values(FormattingCommand)
+
+/**
+ * This is a utility for creating opening and closing markup tag.
+ */
+const createTag = (tags: string) => ({
+  open: `<${tags}>`,
+  close: `</${tags}>`,
+})
+
+const tags = {
+  bold: createTag('b'),
+  italic: createTag('i'),
+  underline: createTag('u'),
+  strikethrough: createTag('strike'),
+}
+
+/**
+ * This determines which commands (bold, italic, underline, strikethrough)
+ * apply to an entire thought string.
+ */
+const getThoughtCommands = (thought: string): CommandState => {
+  // Special case to return early for empty thoughts, treated as having no commands applied
+  if (thought.length === 0) {
+    return {
+      bold: false,
+      italic: false,
+      underline: false,
+      strikethrough: false,
+    }
+  }
+  // Tracks which commands have applied to the entire thought so far
+  const matches: CommandState = {
+    bold: true,
+    italic: true,
+    underline: true,
+    strikethrough: true,
+  }
+  // Tracks which commands apply at the cursor while walking through the thought
+  const cursor: CommandState = {
```

Comment:
Maybe `currentCommandState`?

`cursor` is a specific concept in em, referring to `state.cursor`.

Reply:
Agreed. Let's avoid overloading terms. Changed to `currentCommandState`: f8b8adfb9a74d38e9168d060ae7f71c82aea8fd9